src/TabPanel/TabPanel.tsx

```diff
@@ -0,0 +1,70 @@
+import { Box, Tab, Tabs } from "@mui/material";
+import React, { useEffect, useState } from "react";
+
+import { TabPanelProps } from "./TabPanel.types";
+
+/**
+ * A component for rendering a tab panel
+ *
+ * @param props - The props for the tab panel component
+ * @property children - The children of the tab panel
+ * @property active - The active tab index
+ * @returns The tab panel component
+ */
+const TabPanel = ({ children, active }: TabPanelProps) => {
+  // state for the active tab
+  const [activeTab, setActiveTab] = useState(active ?? 0);
+
+  // update internal state when the active prop changes
+  useEffect(() => {
+    setActiveTab(active ?? 0); // Update internal state when the active prop changes
+  }, [active]);
+
+  // handle tab change
+  const handleChange = (event: React.SyntheticEvent, tab: number) => {
+    setActiveTab(tab);
+  };
+
+  return (
+    <Box>
+      <Tabs
+        value={activeTab}
+        onChange={handleChange}
```

Comment:
Should there be a callback on tab change with clicked tab index so then the app can use it?

Reply:
I didn't think there needs to be, the tab change will just change the view, there isn't anything the parent will do with just a tab change, and anything that happens within that tab will be dealt with within the embed children. I couldn't think of a reason why we need it, if you do I can still add it?

Reply:
As dicussed please add the callback then app will be able to use the index for various purpose , such as to keep the tabe active on reload.